linera-views/src/hashable_wrapper.rs

```diff
@@ -74,26 +68,27 @@ where
         *self.hash.get_mut() = self.stored_hash;
     }
 
-    fn flush(&mut self, batch: &mut Batch) -> Result<(), ViewError> {
-        if self.delete_storage_first() {
-            let mut key_prefix = self.inner.context().base_key();
-            key_prefix.pop();
-            batch.delete_key_prefix(key_prefix);
-            self.stored_hash = None;
-        }
-        self.inner.flush(batch)?;
-        let hash = *self.hash.get_mut();
-        if self.stored_hash != hash {
+    fn flush(&mut self, batch: &mut Batch) -> Result<bool, ViewError> {
+        let test = self.inner.flush(batch)?;
+        if test {
             let mut key = self.inner.context().base_key();
             let tag = key.last_mut().unwrap();
             *tag = KeyTag::Hash as u8;
-            match hash {
-                None => batch.delete_key(key),
-                Some(hash) => batch.put_key_value(key, &hash)?,
+            batch.delete_key(key)
```

Comment:
Actually, would it be better to just `pop()` the last byte? That would supersede the existing delete, and make the batch consist of only a single delete, wouldn't it?

Reply:
Yes, we can do that. But the performance depends on the storage solution and will anyway be hard to measure.